stores/moderationSubmissionsListStore.ts

```diff
@@ -0,0 +1,249 @@
+import { gql } from 'graphql-request'
+import { defineStore } from 'pinia'
+import { ref, type Ref } from 'vue'
+import { gqlClient } from '../utils/graphql.js'
+import type { Submission, Facility, HealthcareProfessional } from '~/typedefs/gqlTypes.js'
+
+export enum SelectedSubmissionListViewTab {
+    ForReview = 'FOR_REVIEW',
+    Approved = 'APPROVED',
+    Rejected = 'REJECTED'
+}
+
+export enum SelectedModerationListView {
+    Facilities = 'FACILITIES',
+    HealthcareProfessionals = 'HEALTHCARE_PROFESSIONALS',
+    Submissions = 'SUBMISSIONS'
+}
+
+export const useModerationSubmissionsListStore = defineStore(
+    'submissionsListStore',
+    () => {
+        const submissionsData: Ref<Submission[]> = ref([])
+        const healthcareProfessionalsData: Ref<HealthcareProfessional[]>
+        = ref([])
+        const facilityData: Ref<Facility[]> = ref([])
+        const selectedModerationListViewChosen: Ref<SelectedModerationListView> = ref(SelectedModerationListView.Submissions)
+        const selectedSubmissionId: Ref<string> = ref('')
+        const selectedFacilityId: Ref<string> = ref('')
+        const selectedHealthcareProfessionalId: Ref<string> = ref('')
+        const selectedSubmissionData: Ref<Submission | undefined> = ref()
+        const filteredSubmissionDataForListComponent: Ref<Submission[]> = ref([])
+
+        async function getSubmissions() {
+            const submissionsSearchResults = await querySubmissions()
+            submissionsData.value = submissionsSearchResults
+        }
+
+        async function getFacilities() {
+            const facilityResults = await queryFacilities()
+            facilityData.value = facilityResults
+        }
+
+        async function getHealthcareProfessionals() {
+            const healthcareProfessionalResults = await queryHealthcareProfessionals()
+            healthcareProfessionalsData.value = healthcareProfessionalResults
+        }
+
+        function setSelectedModerationListViewChosen(selectedOption: SelectedModerationListView) {
+            selectedModerationListViewChosen.value = selectedOption
+        }
+
+        function filterSelectedSubmission(submissionId: string | undefined) {
+            selectedSubmissionData.value = submissionsData.value.find(submission => submission.id === submissionId)
+        }
+
+        function filterSubmissionsBySelectedTab(tabValue: SelectedSubmissionListViewTab) {
+            switch (tabValue) {
+                case SelectedSubmissionListViewTab.ForReview:
+                    filteredSubmissionDataForListComponent.value = submissionsData.value
+                        .filter((submission: Submission) => {
+                            const isNewSubmission = !submission.isRejected && !submission.isApproved && !submission.isUnderReview
+                            return submission.isUnderReview || isNewSubmission
```

Comment:
checking `isUnderReview` twice

Reply:
This is written this way to include if it is under review or if it is completely new and has never even been updated. By default they are all `false` because they haven't been checked at all. But I want both to be filtered into the array. Whether it is new or has been updated but not approved or rejected